# Assignment – High Frequency Words

Group 4
- Santosh Cheruku
- Vinicio Haro
- Javern Wilson
- Saayed Alam

## Problem Statement 

1. Choose a corpus of interest.
2. How many total unique words are in the corpus? (Please feel free to define unique words in any interesting,
defensible way).
3. Taking the most common words, how many unique words represent half of the total words in the corpus?
4. Identify the 200 highest frequency words in this corpus.
5. Create a graph that shows the relative frequency of these 200 words.
6. Does the observed relative frequency of these words follow Zipf’s law? Explain.
7. In what ways do you think the frequency of the words in this corpus differ from “all words in all corpora.”

In [2]:
#Download required packages
import nltk, re, pprint
from nltk import word_tokenize
from nltk.corpus import stopwords
from urllib import request
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

# 1. Choose a Corpus of Interest

Project Guttenberg is a site that allows you to read books directly from the HTML or as a standard txt file on the browser. Because of this, it is an ideal to use as a corpus because we do not need to download data locally. We can pull data directly from the URL thus making this work reproducible. 

For our corpus, we will be using the book BEALBY by H. G. Wells.

http://www.gutenberg.org/files/59769/59769-0.txt

In [4]:
url = "http://www.gutenberg.org/files/59769/59769-0.txt"

response = request.urlopen(url)

raw = response.read().decode('utf8')

## Data cleaning 

The raw text file contains the parts of the book that we do not want to use in our analysis. We do not want to use the publication page nor the chapters page. Thankfully we can impliment methodology to choose where and when to start reading the text into our notebook. 

We will also use the word_tokenize function to convert our raw text into a tokenized version.
https://kite.com/python/docs/nltk.tokenize.word_tokenize

In [8]:
starting_position = raw.find("CHAPTER I")

ending_position = raw.find("End of the Project Gutenberg EBook of Bealby; A Holiday, by H. G. Wells")

raw = raw[starting_position:ending_position]

tokens = word_tokenize(raw)

words = [w for w in tokens if w.isalpha()]

words = [w.lower() for w in words]

len(tokens)

83914

Once we have converted the text into tokens, we want to look at the the number of words. In our case we have 83,914 tokenized words. Converting to tokens will make text operations much more efficient. It should be a data prep step when preparing raw data for some downstream analysis. 

## 2. How many total unique words are in the corpus? (Please feel free to define unique words in any interesting, defensible way).